dataset/dataset_operations_common.h

```diff
@@ -15,10 +15,10 @@ namespace scipp::dataset {
 template <class T1, class T2> auto union_(const T1 &a, const T2 &b) {
   std::map<typename T1::key_type, typename T1::mapped_type> out;
 
-  for (const auto &[key, item] : a)
+  for (const auto [key, item] : a)
     out.emplace(key, item);
 
-  for (const auto &item : b) {
+  for (const auto item : b) {
```

Comment:
Is this what you said might be due to a bad warning? What was the warning, was it specific to the type of b?